### PR TITLE
Fobnail phase 8

### PR DESCRIPTION
**Pull request by arturkow2000**

### CODE REVIEW

**Review by artur-rs — APPROVED** [review]

**Comment by artur-rs** [comment]
@TomaszAIR let's wait for 1 more reviewer

File: blog/content/post/2022-06-17-fobnail_minimal_os.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+| Xous    | No (0)           | No (0)           | Yes (+2)          | No (0)            | Limited (-1)   | No (0)                  | No (0)       | Yes (+1)    | RISC-V only (-1)         | No (0)          |  OK (0)  | 1     |
+| seL4    | No (0) [^1]      | No (0) [^2]      | Yes (+2)          | No (0)            | Limited (-1)   | No (0)                  | Yes (+2)     | Yes (+1)    | Good (+1) [^3]           | Yes (+2)        | OK (but problematic with Genode) (0)  | 7     |
+| Linux   | Yes (+2)         | Yes (+2)         | Yes (+2)          | Yes (+2)          | Yes (+1)       | Yes (kexec) (+2)        | Yes (+2)     | No (0)      | Good (+1) [^5]           | Yes (+2)        |  OK (0)  | 16    |
+| LK      | No (0)           | No (0)           | Limited (-2) [^7] | No (0)            | Yes (+1)       | No (0)                  | Limited (-2) | No (0)      | Good (+1) [^8]           | No (0)          |  OK (0)  | -2    |
+| Fuchsia | Limited (0) [^9] | No (0)           | Yes (+2)          | Limited (0) [^10] | Yes (+1)       | Yes (mexec) (+2)        | Yes (+2)     | Yes (+1)    | Good (+1) [^11]          | No (0)          |  OK (0)  | 7     |
```
Comment (krystian-hebel) [inline blog/content/post/2022-06-17-fobnail_minimal_os.md:83]: There are leftovers from links to footnotes that are not described here, I think they should be removed.

File: blog/content/post/2022-06-17-fobnail_minimal_os.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+## Why Linux?
+
+We have decided to use Linux for building minimal OS because it already has
+everything we need, and other OSes would require a significant amount of work.
+In the future, we may use another OS.
+
+## Reference minimal OS for Fobnail Project
+
+We build a minimal OS image for Fobnail Project by using [Yocto
+Project](https://www.yoctoproject.org/) our
```
Comment (krystian-hebel) [inline blog/content/post/2022-06-17-fobnail_minimal_os.md:99]:
```suggestion
Project](https://www.yoctoproject.org/), our
```

File: blog/content/post/2022-06-17-fobnail_minimal_os.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+[meta-fobnail](https://github.com/fobnail/meta-fobnail) layer and
+[kas](https://github.com/siemens/kas) container. It is based on
+[TrenchBoot](https://trenchboot.org/) project and releated meta layers:
+[meta-pcengines](https://github.com/3mdeb/meta-pcengines) to provide board
+support package for PCengines apu2 and
```
Comment (krystian-hebel) [inline blog/content/post/2022-06-17-fobnail_minimal_os.md:104]:
```suggestion
support package for PC Engines apu2 and
```

File: blog/content/post/2022-06-17-fobnail_minimal_os.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+got from [TrenchBoot](https://github.com/TrenchBoot/) repositories.
+
+We also add [Fobnail Attester](https://github.com/fobnail/fobnail-attester)
+application with required dependencies: `qcbor` and `libcoap`. The application
+itself is automatically started on system startup. After positive result of
+attestation, we can run the target system from an external device by using
+`kexec`.
+
+The process of generating minimal OS is not complicated. In our case, we were
+using a PC running Ubuntu 20.04 Firstly, we need to download the latest
```
Comment (krystian-hebel) [inline blog/content/post/2022-06-17-fobnail_minimal_os.md:122]:
```suggestion
using a PC running Ubuntu 20.04. Firstly, we need to download the latest
```

File: blog/content/post/2022-06-17-fobnail_minimal_os.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+Received message: metadata
+MAC:  0: D:B9:53:D2:50
+SMBIOSv2
+Manufacturer >PC Engines<
+Product Name >apu2<
+Serial Number >1373268<
+Received message: quote
+```
+
+5. Attestation was finished successfully when green LED blinked on Fobnail
```
Comment (krystian-hebel) [inline blog/content/post/2022-06-17-fobnail_minimal_os.md:217]: Blinking reports successful provisioning, attestation result is reported by a constant light for 10 seconds.

File: blog/content/post/2022-06-17-fobnail_minimal_os.md [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+![Fobnail token LEDs](/img/token-led.png)
+
+
+6. Now we can assume that we are in secure environment, so we will execute
+   Ubuntu from external memory by using `kexec`
+
+```
+# mkdir /mnt/usb
+# mount /dev/sda /mnt/usb
+mount: /mnt/usb: WARNING: source write-protected, mounted read-only.
```
Comment (krystian-hebel) [inline blog/content/post/2022-06-17-fobnail_minimal_os.md:229]: Can we pass an option to mount it read-only so this warning is not printed?

**Review by krystian-hebel — CHANGES_REQUESTED** [review]

File: blog/content/post/2022-06-17-fobnail_minimal_os.md (hunk 1 above)
Comment (TomaszAIR) [reply blog/content/post/2022-06-17-fobnail_minimal_os.md]: fixed https://github.com/3mdeb/news-and-ideas/pull/496/commits/acea6f8ce69e7cd420e09461d14ba223d9597e59#diff-5f2030137d33048b104fbebb6f1f3ce21c3e5522e57c8841c0fc8be5ceb88e2cL78

File: blog/content/post/2022-06-17-fobnail_minimal_os.md (hunk 2 above)
Comment (TomaszAIR) [reply blog/content/post/2022-06-17-fobnail_minimal_os.md]: fixed https://github.com/3mdeb/news-and-ideas/pull/496/commits/acea6f8ce69e7cd420e09461d14ba223d9597e59#diff-5f2030137d33048b104fbebb6f1f3ce21c3e5522e57c8841c0fc8be5ceb88e2cL78

File: blog/content/post/2022-06-17-fobnail_minimal_os.md (hunk 4 above)
Comment (TomaszAIR) [reply blog/content/post/2022-06-17-fobnail_minimal_os.md]: fixed https://github.com/3mdeb/news-and-ideas/pull/496/commits/acea6f8ce69e7cd420e09461d14ba223d9597e59#diff-5f2030137d33048b104fbebb6f1f3ce21c3e5522e57c8841c0fc8be5ceb88e2cL122

File: blog/content/post/2022-06-17-fobnail_minimal_os.md (hunk 5 above)
Comment (TomaszAIR) [reply blog/content/post/2022-06-17-fobnail_minimal_os.md]: changed https://github.com/3mdeb/news-and-ideas/pull/496/commits/acea6f8ce69e7cd420e09461d14ba223d9597e59#diff-5f2030137d33048b104fbebb6f1f3ce21c3e5522e57c8841c0fc8be5ceb88e2cL217

Comment (artur-rs) [reply blog/content/post/2022-06-17-fobnail_minimal_os.md]: changed sentence: https://github.com/3mdeb/news-and-ideas/pull/496/commits/9ce86f0f8c8e68faa0fb97df8cf07865b921ea22#diff-470addb504c37cdd533976b2247167df2c35b0e4b11b68ffa5c03fea7ffcd820L217

**Review by krystian-hebel — APPROVED** [review]

**Pull request merged**